app/graphql/resolvers/surveys_search.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require "search_object/plugin/graphql"
+require "graphql/query_resolver"
+
+class Resolvers::SurveysSearch
+  include SearchObject.module(:graphql)
+
+  # NOTE: following line is needed by the SearchObject module but cannot be used in `fetch_results`
+  scope { Survey::Poll.filtered_for_current_user(context[:current_user]) }
+
+  type types[Types::QueryTypes::SurveyPollType]
+
+  option :ids, type: types[types.ID]
+
+  # https://github.com/nettofarah/graphql-query-resolver
+
+  def fetch_results
+    # NOTE: scope from class is not available here, so we need to create a local scope here :/
+    scope = Survey::Poll.filtered_for_current_user(context[:current_user])
+
+    # select only available ids
+    scope = scope.where(id: ids) if ids.present? && ids.any?
+
+    {
+      all: scope.all,
```

Comment:
ist `active` + `archived` nicht ausreichen?
wieso brauchen wir noch `all`?

Reply:
damit das cms alle abfragen kann unabhängig von aktiv oder archiviert.
und damit die app eine bestimmte survey abfragen kann, ohne zu wissen, ob die aktiv ist oder archiviert.

Reply:
aber in `all´ steckt, active und archived mit drin, wir machen also 3 Abfragen an die DB von denen 2 überflüssig sind.

Zudem willst du doch gar nicht alle, denke ich: diejenigen die in der Zukunft liegen sollen in der App nicht liegen

Reply:
:active und :archived sollten doch eigentlich eher Filteroptionen sein wie sonst Beispielswiese CategoryId und ähnliches oder?

Reply:
> - it is tested, that only requested attributes are resolved and processed
>    - if not requesting archived for example, rails is not firing database requests for that

wenn man was davon nicht abfragt, dann wird es auch rails/db seitig nicht requested.
es sind also nur soviele anfragen wie in der query angegeben.

im cms will man alle haben, unabhängig von aktiv oder archiviert.
in der app will man in der übersicht aufgeteilt in aktiv und archiviert.
in der app will man in einem detail screen eine bestimmte per ID, an der stelle kann man günstiger weise über alle gehen, weil da nicht klar ist ob aktiv oder archiviert.

@digorath macht dann in der app z.b.:
![Screenshot 2021-06-16 at 17 30 27](https://user-images.githubusercontent.com/1942953/122356380-0949d180-cf53-11eb-83ee-d8213cb25ea2.png)

Reply:
aber dann geht die Logic nicht in fetch_results rein oder? da wird doch immer ein statischer Hash mit allen 3 Scopes zurückgegeben, und somit tauchen in der DB 3 Calls.

meiner Meinung müsste das irgendwie so aussehen:


```
# frozen_string_literal: true

require "search_object/plugin/graphql"
require "graphql/query_resolver"

class Resolvers::SurveysSearch
  include SearchObject.module(:graphql)

  # NOTE: following line is needed by the SearchObject module but cannot be used in `fetch_results`
  scope { Survey::Poll.filtered_for_current_user(context[:current_user]) }

  type types[Types::QueryTypes::SurveyPollType]

  option :ids, type: types[types.ID], with: :apply_ids
  option :active, type: Boolean, with: :apply_active_filter
  option :archived, type: Boolean, with: :apply_archived_filter

  def apply_ids(scope, value)
    scope.where(id: value)
  end

  def apply_active_filter
    #todo
    # scope.where()
  end

  def apply_archived_filter
    #todo
    # scope.where()
  end

  # https://github.com/nettofarah/graphql-query-resolver

  def fetch_results
    # NOTE: Don't run QueryResolver during tests
    return super unless context.present?

    GraphQL::QueryResolver.run(SurveyPoll, context, Types::QueryTypes::SurveyPollType) do
      super
    end
  end
end

```

Reply:
> da wird doch immer ein statischer Hash mit allen 3 Scopes zurückgegeben, und somit tauchen in der DB 3 Calls.

teste es bitte. hab doch gesagt habe geprüft, dass wenn man nur `all` abfragt mit unserem resolver auch keine anfrage für active und archived rausgeht.

Reply:
wie in slack diskutiert, haben wir dann das problem, dass es mehrere anfragen braucht für versch filter. es können keine aktiven und archivierten surveys gleichzeitig abgerufen werden.

Reply:
@digorath schreibt:
> eigentlich müsste ich es auch so mit einem query abbilden können (eventuell ist noch syntax iwo falsch)
> ![Screenshot 2021-06-17 at 10 19 47](https://user-images.githubusercontent.com/1942953/122359510-e66cec80-cf55-11eb-9df7-ff007ce4f3a2.png)

Reply:
sollte klappen, müsste im gleichen resultieren. weiterhin eine anfrage aber bestehend aus zwei queries.
können wir von mir aus auch so machen. ohne filter dann fürs CMS.

Reply:
geht gut

<img width="1042" alt="Bildschirmfoto 2021-06-17 um 10 49 32" src="https://user-images.githubusercontent.com/1942953/122364104-c63f2c80-cf59-11eb-89b4-513cdcbad524.png">